File: symmetric_encryption/caesar_cipher/caesar_cipher.py
chars : str = "abcdefghijklmnopqrstuvwxyz"

def encryptCeasar(message : str, key : int) -> str:
    encryptedMessage : str = ""

    for char in message:
        encryptedMessage += chars[chars.find(char.lower()) + key % 26]

    return encryptedMessage

def decryptCaesar(message : str, key : int) -> str:
    decryptedMessage : str = ""

    for char in message:
        decryptedMessage += chars[chars.find(char.lower()) - key % 26]

    return decryptedMessage

print(encryptCeasar("hello", 5))

File: symmetric_encryption/caesar_cipher/caesar_cipher.cpp
#include "caesar_cipher.h"

std::string encryptCaesar(int key, std::string message) {
    std::string encryptedMessage = "";
    int charIntValue;

    for(char letter : message) {
        charIntValue = (static_cast<int>(std::tolower(letter)) + key - 97) % 26 + 97;
        encryptedMessage.push_back(static_cast<char>(charIntValue));
    }

    return encryptedMessage;
}

std::string decryptCaesar(int key, std::string message) {
    std::string decryptedMessage = "";
     int charIntValue;

    for(char letter : message) {
        charIntValue = (static_cast<int>(std::tolower(letter)) - key - 97) % 26 + 97;
        decryptedMessage.push_back(static_cast<char>(charIntValue));
    }

    return decryptedMessage;
}
